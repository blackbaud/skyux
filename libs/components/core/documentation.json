{
  "$schema": "../manifest/documentation-schema.json",
  "groups": {
    "id": {
      "docsIds": [
        "SkyIdModule",
        "SkyIdDirective",
        "SkyIdService",
        "CoreIdExampleComponent"
      ],
      "primaryDocsId": "SkyIdModule"
    },
    "media-query": {
      "docsIds": [
        "SkyMediaQueryService",
        "SkyMediaBreakpoints",
        "SkyMediaQueryListener",
        "SkyBreakpoint",
        "provideSkyMediaQueryTesting",
        "SkyMediaQueryTestingController",
        "CoreMediaQueryBasicExampleComponent",
        "CoreMediaQueryResponsiveHostExampleComponent"
      ],
      "primaryDocsId": "SkyMediaQueryService"
    },
    "numeric": {
      "docsIds": [
        "SkyNumericModule",
        "SkyNumericPipe",
        "SkyNumericService",
        "SkyNumericOptions",
        "CoreNumericBasicExampleComponent"
      ],
      "primaryDocsId": "SkyNumericModule"
    }
  }
}
