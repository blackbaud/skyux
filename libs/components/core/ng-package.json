{
  "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../dist/libs/components/core",
  "lib": {
    "entryFile": "src/index.ts",
    "styleIncludePaths": ["../../.."]
  },
  "inlineStyleLanguage": "scss"
}
