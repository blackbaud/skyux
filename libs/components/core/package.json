{
  "name": "@skyux/core",
  "version": "0.0.0-PLACEHOLDER",
  "author": "Blackbaud, Inc.",
  "keywords": [
    "blackbaud",
    "skyux"
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/blackbaud/skyux.git"
  },
  "bugs": {
    "url": "https://github.com/blackbaud/skyux/issues"
  },
  "homepage": "https://github.com/blackbaud/skyux#readme",
  "peerDependencies": {
    "@angular/common": "^13.3.2",
    "@angular/core": "^13.3.2",
    "@angular/platform-browser": "^13.3.2",
    "@angular/router": "^13.3.2",
    "@skyux/i18n": "0.0.0-PLACEHOLDER"
  },
  "dependencies": {
    "tslib": "^2.3.1"
  }
}
