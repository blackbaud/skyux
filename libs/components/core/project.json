{
  "name": "core",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/components/core/src",
  "prefix": "sky",
  "targets": {
    "build": {
      "executor": "@nx/angular:package",
      "outputs": ["{workspaceRoot}/dist/libs/components/core"],
      "options": {
        "project": "libs/components/core/ng-package.json",
        "updateBuildableProjectDepsInPackageJson": false
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/components/core/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/components/core/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@angular-devkit/build-angular:karma",
      "options": {
        "tsConfig": "libs/components/core/tsconfig.spec.json",
        "karmaConfig": "libs/components/core/karma.conf.js",
        "styles": [
          "libs/components/theme/src/lib/styles/sky.scss",
          "libs/components/theme/src/lib/styles/themes/modern/styles.scss"
        ],
        "codeCoverage": true,
        "codeCoverageExclude": ["**/fixtures/**"],
        "polyfills": [
          "zone.js",
          "zone.js/testing",
          "libs/components/packages/src/polyfills.js"
        ]
      }
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/components/core/src/**/*.ts",
          "libs/components/core/src/**/*.html"
        ]
      }
    }
  },
  "tags": ["component", "npm"]
}
