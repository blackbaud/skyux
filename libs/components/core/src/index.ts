export * from './lib/modules/adapter-service/adapter.module';
export * from './lib/modules/adapter-service/adapter.service';
export * from './lib/modules/adapter-service/focusable-children-options';

export * from './lib/modules/affix/affix-auto-fit-context';
export * from './lib/modules/affix/affix-config';
export * from './lib/modules/affix/affix-horizontal-alignment';
export * from './lib/modules/affix/affix-offset';
export * from './lib/modules/affix/affix-offset-change';
export * from './lib/modules/affix/affix-placement';
export * from './lib/modules/affix/affix-placement-change';
export * from './lib/modules/affix/affix-vertical-alignment';
export * from './lib/modules/affix/affix.module';
export * from './lib/modules/affix/affix.service';
export * from './lib/modules/affix/affixer';

export * from './lib/modules/dock/dock-insert-component-config';
export * from './lib/modules/dock/dock-item';
export * from './lib/modules/dock/dock-item-config';
export * from './lib/modules/dock/dock-location';
export * from './lib/modules/dock/dock-options';
export * from './lib/modules/dock/dock.module';
export * from './lib/modules/dock/dock.service';

export * from './lib/modules/dynamic-component/dynamic-component-location';
export * from './lib/modules/dynamic-component/dynamic-component-options';
export * from './lib/modules/dynamic-component/dynamic-component.module';
export * from './lib/modules/dynamic-component/dynamic-component.service';

export * from './lib/modules/format/app-format';

export * from './lib/modules/id/id.module';

export * from './lib/modules/log/log.module';
export * from './lib/modules/log/log.service';

export * from './lib/modules/media-query/media-breakpoints';
export * from './lib/modules/media-query/media-query-listener';
export * from './lib/modules/media-query/media-query.module';
export * from './lib/modules/media-query/media-query.service';

export * from './lib/modules/mutation/mutation-observer-service';

export * from './lib/modules/numeric/numeric-symbol';
export * from './lib/modules/numeric/numeric.module';
export * from './lib/modules/numeric/numeric.options';
export * from './lib/modules/numeric/numeric.pipe';
export * from './lib/modules/numeric/numeric.service';

export * from './lib/modules/overlay/overlay-config';
export * from './lib/modules/overlay/overlay-instance';
export * from './lib/modules/overlay/overlay.module';
export * from './lib/modules/overlay/overlay.service';

export * from './lib/modules/percent-pipe/percent-pipe.module';
export * from './lib/modules/percent-pipe/percent.pipe';

export * from './lib/modules/resize-observer/resize-observer.service';
export * from './lib/modules/resize-observer/resize-observer-media-query.service';

export * from './lib/modules/scrollable-host/scrollable-host.service';

export * from './lib/modules/title/set-title-args';
export * from './lib/modules/title/title.service';

export * from './lib/modules/ui-config/ui-config.service';

export * from './lib/modules/viewkeeper/viewkeeper-host-options';
export * from './lib/modules/viewkeeper/viewkeeper-options';
export * from './lib/modules/viewkeeper/viewkeeper.module';
export * from './lib/modules/viewkeeper/viewkeeper.service';

export * from './lib/modules/window/window-ref';

// Components and directives must be exported to support Angular's "partial" Ivy compiler.
// Obscure names are used to indicate types are not part of the public API.
export { SkyAffixDirective as λ1 } from './lib/modules/affix/affix.directive';
export { SkyIdDirective as λ2 } from './lib/modules/id/id.directive';
export { SkyViewkeeperDirective as λ3 } from './lib/modules/viewkeeper/viewkeeper.directive';
