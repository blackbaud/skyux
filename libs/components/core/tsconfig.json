{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "types": []
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.lib.prod.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
