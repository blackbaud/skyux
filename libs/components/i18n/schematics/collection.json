{
  "$schema": "../../../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "ng-add": {
      "description": "Add @skyux/i18n to your project",
      "factory": "./ng-add/ng-add.schematic"
    },
    "lib-resources-module": {
      "description": "Generate library resources module",
      "factory": "./ng-generate/lib-resources-module/lib-resources-module.schematic",
      "schema": "./ng-generate/lib-resources-module/schema.json"
    }
  }
}
