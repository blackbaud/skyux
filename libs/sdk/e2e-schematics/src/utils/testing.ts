import {
  E2eTestRunner,
  UnitTestRunner,
  applicationGenerator,
  libraryGenerator,
} from '@nx/angular/generators';
import { Tree } from '@nx/devkit';
import { Linter } from '@nx/eslint';

export async function createTestApplication(
  tree: Tree,
  options: { name: string; e2eTestRunner?: boolean; unitTestRunner?: boolean },
): Promise<void> {
  await applicationGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `apps/${options.name}`,
    e2eTestRunner: options.e2eTestRunner
      ? E2eTestRunner.Cypress
      : E2eTestRunner.None,
    linter: Linter.None,
    unitTestRunner: options.unitTestRunner
      ? UnitTestRunner.Jest
      : UnitTestRunner.None,
  });
}

export async function createTestLibrary(
  tree: Tree,
  options: { name: string },
): Promise<void> {
  await libraryGenerator(tree, {
    name: options.name,
    skipPackageJson: true,
    directory: `libs/${options.name}`,
    linter: Linter.None,
    unitTestRunner: UnitTestRunner.None,
  });
}
