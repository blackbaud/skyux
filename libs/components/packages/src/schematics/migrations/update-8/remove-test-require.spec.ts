import { SchematicContext } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';

import { createTestLibrary } from '../../testing/scaffold';

import RemoveTestRequire from './remove-test-require';

const testTsPieces: Record<string, string> = {
  imports: `
import 'zone.js';
import 'zone.js/testing';
import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting
} from '@angular/platform-browser-dynamic/testing';
`,
  fix: `
// Fix for crossvent "global is not defined" error. The crossvent library
// is used by Dragula, which in turn is used by multiple SKY UX components.
// See: https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;
`,
  requirePolyfill: `
declare const require: {
  context(path: string, deep?: boolean, filter?: RegExp): {
    keys(): string[];
    <T>(id: string): T;
  };
};
`,
  getTestBed: `
// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  { teardown: { destroyAfterEach: true }},
);
`,
  context: `
// Then we find all the tests.
const context = require.context('./', true, /\\.spec\\.ts$/);
// And load the modules.
context.keys().map(context);
`,
  testingContext: `
// Find any tests included in the "testing" entry point.
try {
  const testingContext = require.context('../testing/', true, /\\.spec\\.ts$/);
  testingContext.keys().map(testingContext);
} catch (err) {}
`,
};

describe('RemoveTestRequire', () => {
  const runner = new SchematicTestRunner(
    'schematics',
    require.resolve('../migration-collection.json')
  );

  it('should create an instance', async () => {
    const tree = await createTestLibrary(runner, { projectName: 'test' });
    expect(tree.exists('projects/test/src/test.ts')).toBe(false);
    tree.create(
      'projects/test/src/test.ts',
      testTsPieces.imports +
        testTsPieces.fix +
        testTsPieces.requirePolyfill +
        testTsPieces.getTestBed +
        testTsPieces.context +
        testTsPieces.testingContext
    );
    await RemoveTestRequire()(tree, {} as SchematicContext);
    expect(tree.readText('projects/test/src/test.ts')).toContain(
      testTsPieces.imports
    );
    expect(tree.readText('projects/test/src/test.ts')).not.toContain(
      'declare const require'
    );
    expect(tree.readText('projects/test/src/test.ts')).not.toContain(
      'const context = require.context'
    );
    expect(tree.readText('projects/test/src/test.ts')).not.toContain(
      'const testingContext = require.context'
    );
  });

  it('should work without testing context', async () => {
    const tree = await createTestLibrary(runner, { projectName: 'test' });
    expect(tree.exists('projects/test/src/test.ts')).toBe(false);
    tree.create(
      'projects/test/src/test.ts',
      testTsPieces.imports +
        testTsPieces.fix +
        testTsPieces.requirePolyfill +
        testTsPieces.getTestBed +
        testTsPieces.context
    );
    await RemoveTestRequire()(tree, {} as SchematicContext);
    expect(tree.readText('projects/test/src/test.ts')).toContain(
      testTsPieces.imports
    );
    expect(tree.readText('projects/test/src/test.ts')).not.toContain(
      'declare const require'
    );
    expect(tree.readText('projects/test/src/test.ts')).not.toContain(
      'const context = require.context'
    );
  });

  it('should work with no changes', async () => {
    const tree = await createTestLibrary(runner, { projectName: 'test' });
    expect(tree.exists('projects/test/src/test.ts')).toBe(false);
    tree.create(
      'projects/test/src/test.ts',
      testTsPieces.imports + testTsPieces.fix + testTsPieces.getTestBed
    );
    await RemoveTestRequire()(tree, {} as SchematicContext);
    expect(tree.readText('projects/test/src/test.ts')).toContain(
      testTsPieces.imports
    );
  });
});
