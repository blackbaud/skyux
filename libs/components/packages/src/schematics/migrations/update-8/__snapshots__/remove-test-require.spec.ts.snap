// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveTestRequire should create an instance 1`] = `
"
import 'zone.js';
import 'zone.js/testing';
import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting
} from '@angular/platform-browser-dynamic/testing';

// Fix for crossvent \\"global is not defined\\" error. The crossvent library
// is used by Dragula, which in turn is used by multiple SKY UX components.
// See: https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;

// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  { teardown: { destroyAfterEach: true }},
);
"
`;

exports[`RemoveTestRequire should work with no changes 1`] = `
"
import 'zone.js';
import 'zone.js/testing';
import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting
} from '@angular/platform-browser-dynamic/testing';

// Fix for crossvent \\"global is not defined\\" error. The crossvent library
// is used by Dragula, which in turn is used by multiple SKY UX components.
// See: https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;

// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  { teardown: { destroyAfterEach: true }},
);
"
`;

exports[`RemoveTestRequire should work without context map 1`] = `
"
import 'zone.js';
import 'zone.js/testing';
import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting
} from '@angular/platform-browser-dynamic/testing';

// Fix for crossvent \\"global is not defined\\" error. The crossvent library
// is used by Dragula, which in turn is used by multiple SKY UX components.
// See: https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;

// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  { teardown: { destroyAfterEach: true }},
);
"
`;

exports[`RemoveTestRequire should work without testing context 1`] = `
"
import 'zone.js';
import 'zone.js/testing';
import { getTestBed } from '@angular/core/testing';
import {
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting
} from '@angular/platform-browser-dynamic/testing';

// Fix for crossvent \\"global is not defined\\" error. The crossvent library
// is used by Dragula, which in turn is used by multiple SKY UX components.
// See: https://github.com/bevacqua/dragula/issues/602
(window as any).global = window;

// First, initialize the Angular testing environment.
getTestBed().initTestEnvironment(
  BrowserDynamicTestingModule,
  platformBrowserDynamicTesting(),
  { teardown: { destroyAfterEach: true }},
);
"
`;
